hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java

```diff
@@ -766,4 +768,103 @@ public void testTempMount() throws Exception {
         fileStatusesInDir1[0].getPath().toUri().getPath());
   }
 
+  /**
+   * Helper function. Check Ozone volume existence.
+   * @param volumeStr Name of the volume
+   * @return true if volume exists, false if not
+   */
+  private boolean volumeExist(String volumeStr) throws IOException {
+    try {
+      objectStore.getVolume(volumeStr);
+    } catch (OMException ex) {
+      if (ex.getResult() == VOLUME_NOT_FOUND) {
+        return false;
+      } else {
+        throw ex;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Helper function. Delete a path non-recursively and expect failure.
+   * @param f Path to delete.
+   * @throws IOException
+   */
+  private void deleteNonRecursivelyAndFail(Path f) throws IOException {
+    try {
+      fs.delete(f, false);
+      Assert.fail("Should have thrown PathIsNotEmptyDirectoryException!");
+    } catch (PathIsNotEmptyDirectoryException ignored) {
+    }
+  }
+
+  @Test
+  public void testDeleteVolumeAndBucket() throws IOException {
```

Comment:
I think these test cases should be split into multiple cases?

Reply:
Good idea

Reply:
done in https://github.com/apache/hadoop-ozone/pull/906/commits/6d7a21b911b7b1ecff3a60443fa72782c5425de0